cunumeric/module.py

```diff
@@ -6227,3 +6227,719 @@ def bincount(
+        This method gives continuous results using:
+        * alpha = 0
+        * beta = 1
+    hazen:
+        method 5 of H&F [1]_.
+        This method gives continuous results using:
+        * alpha = 1/2
+        * beta = 1/2
+    weibull:
+        method 6 of H&F [1]_.
+        This method gives continuous results using:
+        * alpha = 0
+        * beta = 0
+    linear:
+        method 7 of H&F [1]_.
+        This method gives continuous results using:
+        * alpha = 1
+        * beta = 1
+    median_unbiased:
+        method 8 of H&F [1]_.
+        This method is probably the best method if the sample
+        distribution function is unknown (see reference).
+        This method gives continuous results using:
+        * alpha = 1/3
+        * beta = 1/3
+    normal_unbiased:
+        method 9 of H&F [1]_.
+        This method is probably the best method if the sample
+        distribution function is known to be normal.
+        This method gives continuous results using:
+        * alpha = 3/8
+        * beta = 3/8
+    lower:
+        NumPy method kept for backwards compatibility.
+        Takes ``i`` as the interpolation point.
+    higher:
+        NumPy method kept for backwards compatibility.
+        Takes ``j`` as the interpolation point.
+    nearest:
+        NumPy method kept for backwards compatibility.
+        Takes ``i`` or ``j``, whichever is nearest.
+    midpoint:
+        NumPy method kept for backwards compatibility.
+        Uses ``(i + j) / 2``.
+    References
+    ----------
+    .. [1] R. J. Hyndman and Y. Fan,
+       "Sample quantiles in statistical packages,"
+       The American Statistician, 50(4), pp. 361-365, 1996
+    """
+
+    from .logic import isscalar
+
+    dict_methods = {
+        "inverted_cdf": inverted_cdf,
+        "averaged_inverted_cdf": averaged_inverted_cdf,
+        "closest_observation": closest_observation,
+        "interpolated_inverted_cdf": interpolated_inverted_cdf,
+        "hazen": hazen,
+        "weibull": weibull,
+        "linear": linear,
+        "median_unbiased": median_unbiased,
+        "normal_unbiased": normal_unbiased,
+        "lower": lower,
+        "higher": higher,
+        "midpoint": midpoint,
+        "nearest": nearest,
+    }
+
+    axes_set = []
+    original_shape = a.shape
+
+    if (axis is not None) and (not isscalar(axis)):
+        if len(axis) == 1:
+            real_axis = axis[0]
+            a_rr = a
+            axes_set = [real_axis]
+        else:
+            (real_axis, a_rr) = reshuffle_reshape(a, axis)
+            # What happens with multiple axes and overwrite_input = True ?
+            # It seems overwrite_input is reset to False;
+            overwrite_input = False
+            axes_set = axis
+    else:
+        real_axis = axis
+        a_rr = a
+        axes_set = [real_axis]
+
+    input_is_scalar = isscalar(q)
+
+    if input_is_scalar:
+        q_arr = np.asarray(q)  # 0-dimensional
+    else:
+        q_arr = q
```

Comment:
Let's do `q_arr = np.asarray(q)` unconditionally. Otherwise if the users passes e.g. the quantiles in a list:
```
a = cn.ones((3,4,5))
print(cn.quantile(a, [0.5,0.5], axis=1).shape)
```
then `q_arr.shape` will fail.

Reply:
Done.